### ISSUE TITLE
(python3) installation failing chocolateyInstall.ps1

### ISSUE THREAD

**Issue by bilalanwer24h** [chocolatey-community/chocolatey-packages#1691]
Python should be installed successfully via chocolateyInstall.ps1'.

================================================================
PS C:\Windows\system32> choco install python3
Chocolatey v0.10.15
Installing the following packages:
python3
By installing you accept licenses for the packages.
Progress: Downloading python3 3.10.0-b4... 100%

python3 v3.10.0-b4 [Approved]
python3 package files install completed. Performing other installation steps.
The package python3 wants to run 'chocolateyInstall.ps1'.
Note: If you don't run this script, the installation will fail.
Note: To confirm automatically next time, use '-y' or consider:
choco feature enable -n allowGlobalConfirmation
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint): yes

Installing 64-bit python3...
ERROR: Running ["C:\ProgramData\chocolatey\lib\python3\tools\python-3.10.0b4-amd64.exe" /quiet InstallAllUsers=1 Prepend
Path=1 TargetDir="C:\Python310" ] was not successful. Exit code was '1'. See log for possible error messages.
The install of python3 was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\python3\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - python3 (exited 1) - Error while running 'C:\ProgramData\chocolatey\lib\python3\tools\chocolateyInstall.ps1'.
 See log for details.

========================================================

Any help will be highly appreciated. 

Regards

**Comment by pauby** [comment]
Can you let us know what you've done to troubleshoot it?

**Comment by Mugane** [comment]
Please reopen this. No solution exists as far as I can tell. 

The error says there is a log but no log file location is provided

**Comment by pauby** [comment]
Works fine here in Windows 10 sandbox. I think this is a local environment issue.

```
PS C:\Users\WDAGUtilityAccount> choco install -y python
Chocolatey v0.12.1
Installing the following packages:
python
By installing, you accept licenses for the packages.
Progress: Downloading python3 3.10.2... 100%
Progress: Downloading KB2919355 1.0.20160915... 100%
Progress: Downloading python 3.10.2... 100%

chocolatey-core.extension v1.3.5.1
chocolatey-core.extension package files install completed. Performing other installation steps.
 Installed/updated chocolatey-core extensions.
 The install of chocolatey-core.extension was successful.
  Software installed to 'C:\ProgramData\chocolatey\extensions\chocolatey-core'

chocolatey-windowsupdate.extension v1.0.4
chocolatey-windowsupdate.extension package files install completed. Performing other installation steps.
 Installed/updated chocolatey-windowsupdate extensions.
 The install of chocolatey-windowsupdate.extension was successful.
  Software installed to 'C:\ProgramData\chocolatey\extensions\chocolatey-windowsupdate'

KB3035131 v1.0.3
kb3035131 package files install completed. Performing other installation steps.
Skipping installation because update KB3035131 does not apply to this operating system (Microsoft Windows 10 Enterprise).
 The install of kb3035131 was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.

KB3033929 v1.0.5
kb3033929 package files install completed. Performing other installation steps.
Skipping installation because update KB3033929 does not apply to this operating system (Microsoft Windows 10 Enterprise).
 The install of kb3033929 was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.

KB2919442 v1.0.20160915
kb2919442 package files install completed. Performing other installation steps.
Skipping installation because this hotfix only applies to Windows 8.1 and Windows Server 2012 R2.
 The install of kb2919442 was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.

KB2919355 v1.0.20160915 [Approved]
kb2919355 package files install completed. Performing other installation steps.
Skipping installation because this hotfix only applies to Windows 8.1 and Windows Server 2012 R2.
 The install of kb2919355 was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.

KB2999226 v1.0.20181019
kb2999226 package files install completed. Performing other installation steps.
Skipping installation because update KB2999226 does not apply to this operating system (Microsoft Windows 10 Enterprise).
 The install of kb2999226 was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.

vcredist140 v14.28.29913
vcredist140 package files install completed. Performing other installation steps.
Installing vcredist140-x86...
vcredist140-x86 has been installed.
Installing vcredist140-x64...
vcredist140-x64 has been installed.
  vcredist140 may be able to be automatically uninstalled.
 The install of vcredist140 was successful.
  Software installed as 'exe', install location is likely default.

vcredist2015 v14.0.24215.20170201
vcredist2015 package files install completed. Performing other installation steps.
 The install of vcredist2015 was successful.
  Software installed to 'C:\ProgramData\chocolatey\lib\vcredist2015'

python3 v3.10.2 [Approved]
python3 package files install completed. Performing other installation steps.
Installing 64-bit python3...
python3 has been installed.
Python installed to: 'C:\Python310'
Restricting write permissions to Administrators
  python3 can be automatically uninstalled.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The install of python3 was successful.
  Software installed as 'exe', install location is likely default.

python v3.10.2 [Approved]
python package files install completed. Performing other installation steps.
 The install of python was successful.
  Software installed to 'C:\ProgramData\chocolatey\lib\python'

Chocolatey installed 11/11 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Installed:
 - kb3033929 v1.0.5
 - python3 v3.10.2
 - chocolatey-windowsupdate.extension v1.0.4
 - vcredist140 v14.28.29913
 - kb2999226 v1.0.20181019
 - kb2919355 v1.0.20160915
 - chocolatey-core.extension v1.3.5.1
 - kb2919442 v1.0.20160915
 - vcredist2015 v14.0.24215.20170201
 - kb3035131 v1.0.3
 - python v3.10.2
```

**Comment by Mugane** [comment]
In case it helps anyone, this was a compatibility issue - Python version >3.8 does not work on Windows 7

**Comment by tallpeak** [comment]
(In case anyone else finds this useful) I got the same error; chocolatey was trying to install 3.11.0 when 3.11.1 is already installed. So I used choco pin -n=python to prevent future updates. Now choco upgrade all says ... python v3.11.0 is the latest version available based on your source(s). ... 
===
chocolatey.log tail:
2023-01-13 14:21:06,806 1092 [ERROR] - Failures
2023-01-13 14:21:06,809 1092 [ERROR] -  - python3 (exited 1638) - Error while running 'C:\ProgramData\chocolatey\lib\python3\tools\chocolateyInstall.ps1'.
 See log for details.
2023-01-13 14:21:06,812 1092 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2023-01-13 14:21:07,101 1092 [DEBUG] - Exiting with 1638

rg Error "C:\Users\tallp\AppData\Local\Temp\chocolatey\Python 3.11.0 (64-bit)_20230113141906.log"

171:[2FDC:1B5C][2023-01-13T14:19:07]e000: Error 0x80070666: Cannot install a product when a newer version is installed.

python says:
Python 3.11.1 (tags/v3.11.1:a7a450f, Dec  6 2022, 19:58:39) [MSC v.1934 64 bit (AMD64)] on win32

So I'm trying:
C:\Program Files\smartmontools\bin>choco pin add -n=python
Chocolatey v1.2.1
Trying to add a pin for python
Successfully added a pin for python v3.11.0.

ref: https://superuser.com/questions/1011003/remove-chocolatey-registration-without-uninstalling-application
